pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -4278,16 +4298,33 @@ func mapNodeNameToPrivateDNSName(nodeName types.NodeName) string {
 }
 
 // mapInstanceToNodeName maps a EC2 instance to a k8s NodeName, by extracting the PrivateDNSName
-func mapInstanceToNodeName(i *ec2.Instance) types.NodeName {
-	return types.NodeName(aws.StringValue(i.PrivateDnsName))
+func mapInstanceToNodeName(i *ec2.Instance, explicitNames bool) (types.NodeName, error) {
+	if explicitNames {
+		for _, tag := range i.Tags {
```

Comment:
I.e. check to see if a tag exists that matches `kubernetes.io/nodeName`, and if not, we fall back to private DNS instead of erroring.